blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/credential/BlueOceanCredentialsProvider.java

```diff
@@ -0,0 +1,262 @@
+
+    @Nonnull
+    @Override
+    public <C extends Credentials> List<C> getCredentials(@Nonnull Class<C> type, @Nullable ItemGroup itemGroup,
+                                                          @Nullable Authentication authentication) {
+        if (itemGroup instanceof AbstractFolder) {
+            return getCredentials(type, (AbstractFolder)itemGroup);
+        }
+        return Collections.emptyList();
+    }
+
+    private static <C extends Credentials> List<C> getCredentials(Class<C> type, AbstractFolder folder){
+        FolderPropertyImpl prop = (FolderPropertyImpl)folder.getProperties().get(FolderPropertyImpl.class);
+        if (prop != null) {
+            User user = User.get(prop.getUser(), false, Collections.emptyMap());
+            if(user != null){
+                Authentication auth = user.impersonate();
+                return CredentialsMatchers.filter(
+                        CredentialsProvider.lookupCredentials(
+                                type,
+                                Jenkins.getInstance(),
+                                auth,
+                                Collections.<DomainRequirement>emptyList()
+                        ),
+                        CredentialsMatchers.allOf(CredentialsMatchers.withId(prop.getId()),
+                                CredentialsMatchers.withScope(CredentialsScope.USER))
+                );
+            }
+        }
+        return Collections.emptyList();
+    }
+
+    @Nonnull
+    @Override
+    public <C extends IdCredentials> ListBoxModel getCredentialIds(@Nonnull Class<C> type,
+                                                                   @Nullable ItemGroup itemGroup,
+                                                                   @Nullable Authentication authentication,
+                                                                   @Nonnull List<DomainRequirement> domainRequirements,
+                                                                   @Nonnull CredentialsMatcher matcher) {
+        ListBoxModel result = new ListBoxModel();
+        if (itemGroup instanceof AbstractFolder) {
+            FolderPropertyImpl prop = (FolderPropertyImpl) ((AbstractFolder) itemGroup).getProperties().get(FolderPropertyImpl.class);
+            if (prop != null) {
+                result.add(DISPLAY_NAME, prop.getId());
+            }
+
+        }
+        return result;
+    }
+
+    @Override
+    @Nonnull
+    public String getDisplayName() {
+        return DISPLAY_NAME;
+    }
+
+    @Override
+    public CredentialsStore getStore(@CheckForNull ModelObject object) {
+        if(isApplicable(object)) {
+            return new FolderPropertyImpl.StoreImpl((AbstractFolder) object);
+        }
+        return null;
+    }
+
+    @Override
+    public Set<CredentialsScope> getScopes(ModelObject object) {
+        if(isApplicable(object)){
+            return ImmutableSet.of(CredentialsScope.USER);
```

Comment:
Actually you are `CredentailsScope.GLOBAL` if we are to be strictly correct... this may require changes in the credentials API to allow you to "clone" a credential while changing it's scope